File: docs/versioned_docs/version-1.9.0/API-Reference/javascript-examples/api-files/upload-image-files-v1.js
const fs = require("fs");
const path = require("path");

const fixturesDir = path.join(__dirname, "../../fixtures");
const defaultImage = path.join(fixturesDir, "sample-upload.png");
const imagePath = process.env.SAMPLE_IMAGE_FILE || defaultImage;
const imageBuf = fs.readFileSync(imagePath);
const imageName = path.basename(imagePath);

const url = `${process.env.LANGFLOW_URL ?? ""}/api/v1/files/upload/${process.env.FLOW_ID ?? ""}`;

const formData = new FormData();
formData.append("file", new Blob([imageBuf]), imageName);

const options = {
  method: "POST",
  headers: {
    "x-api-key": `${process.env.LANGFLOW_API_KEY ?? ""}`,
  },
  body: formData,
};

fetch(url, options)
  .then(async (response) => {
    if (!response.ok) {
      throw new Error(`HTTP ${response.status}`);
    }
    const text = await response.text();
    console.log(text);
  })
  .catch((error) => console.error(error));

File: docs/versioned_docs/version-1.11.0/API-Reference/python-examples/api-files/upload-image-files-v1.py
import os

import requests

url = f"{os.getenv('LANGFLOW_URL', '')}/api/v1/files/upload/{os.getenv('FLOW_ID', '')}"

headers = {
    "x-api-key": f"{os.getenv('LANGFLOW_API_KEY', '')}",
}

files = {
    "file": open(os.getenv("SAMPLE_IMAGE_FILE", "docs/docs/API-Reference/fixtures/sample-upload.png"), "rb"),
}

response = requests.request("POST", url, headers=headers, files=files)
response.raise_for_status()

print(response.text)

for _f in files.values():
    if hasattr(_f, "close"):
        _f.close()
